src/tool_operate.c

```diff
@@ -1658,6 +1662,23 @@ static CURLcode single_transfer(struct GlobalConfig *global,
             break;
         }
 
+#ifdef CURL_CA_EMBED
+        if(!config->cacert && !config->capath) {
+          struct curl_blob blob;
+          blob.data = (void *)curl_ca_embed;
+          blob.len = strlen((const char *)curl_ca_embed);
+          blob.flags = CURL_BLOB_NOCOPY;
+          curl_easy_setopt(curl, CURLOPT_CAINFO_BLOB, &blob);
+        }
+        if(!config->proxy_cacert && !config->proxy_capath) {
+          struct curl_blob blob;
+          blob.data = (void *)curl_ca_embed;
+          blob.len = strlen((const char *)curl_ca_embed);
+          blob.flags = CURL_BLOB_NOCOPY;
+          curl_easy_setopt(curl, CURLOPT_PROXY_CAINFO_BLOB, &blob);
+        }
+#endif
```

Comment:
You should check the return codes from the setopt() calls.

As [our TLS option table shows](https://curl.se/libcurl/c/tls-options.html), these options are not supported by all TLS backends so there's a risk someone builds with this but then it fails at run-time because libcurl cannot comply!

Reply:
Good idea to catch it here indeed.